**Условие 1: Задача 1**
ФАЙЛ С ДОМАМИ ИЗ ПРОШЛОГО ДЗ
Считать данные с помощью pandas
Вывести на экран первые 5 строк

In [1]:
import pandas as pd
df = pd.read_csv('D:\Lena\Geekbrains\HomeWork\PythonHW\kc_house_data.csv', sep=',')
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

**1.2** Создать новый признак delta_renovated, который будет содержать разницу в годах между годом реновацией дома и годом постройки дома

Если реновации дома не было, то в новом признаке поставьте 0

In [2]:
df['delta_renovated']=df['yr_renovated']-df['yr_built']
df.loc[df['delta_renovated']<0,'delta_renovated']=0
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...        1180              0   
1      7242     2.0           0     0  ...        2170            400   
2     10000     1.0           0     0  ...         770              0   
3      5000     1.0           0     0  ...        1050            910   
4      8080     1.0           0     0  ...        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  delta_renovated  
0        5650                0  
1        7639               40  
2        8062                0  
3        5000                0  
4        7503                0  

[5 rows x 22 columns]

**1.3** Создайте признаки года продажи, месяца продажи

In [3]:
df['yr_sale']= df['date'].apply(lambda x: x [0:4])
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  sqft_basement  yr_built  \
0      5650     1.0           0     0  ...              0      1955   
1      7242     2.0           0     0  ...            400      1951   
2     10000     1.0           0     0  ...              0      1933   
3      5000     1.0           0     0  ...            910      1965   
4      8080     1.0           0     0  ...              0      1987   

   yr_renovated  zipcode      lat     long  sqft_living15  sqft_lot15  \
0             0    98178  47.5112 -122.257           1340        5650   
1          1991    98125  47.7210 -122.319           1690        7639   
2             0    98028  47.7379 -122.233           2720        8062   
3             0    98136  47.5208 -122.393           1360        5000   
4             0    98074  47.6168 -122.045           1800        7503   

   delta_renovated  yr_sale  
0                0     2014  
1               40     2014  
2                0     2015  
3                0     2014  
4                0     2015  

[5 rows x 23 columns]

In [4]:
def month(x):
    mon=int(x[4:6])
    if (mon>=1) & (mon<=12):
        return ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Okt','Nov','Dec'][mon-1]
    else:
        return 'bad mon number!'

df['month_sale']= df['date'].apply(month)
df.head(20)

id             date      price  bedrooms  bathrooms  sqft_living  \
0   7129300520  20141013T000000   221900.0         3       1.00         1180   
1   6414100192  20141209T000000   538000.0         3       2.25         2570   
2   5631500400  20150225T000000   180000.0         2       1.00          770   
3   2487200875  20141209T000000   604000.0         4       3.00         1960   
4   1954400510  20150218T000000   510000.0         3       2.00         1680   
5   7237550310  20140512T000000  1225000.0         4       4.50         5420   
6   1321400060  20140627T000000   257500.0         3       2.25         1715   
7   2008000270  20150115T000000   291850.0         3       1.50         1060   
8   2414600126  20150415T000000   229500.0         3       1.00         1780   
9   3793500160  20150312T000000   323000.0         3       2.50         1890   
10  1736800520  20150403T000000   662500.0         3       2.50         3560   
11  9212900260  20140527T000000   468000.0         2       1.00         1160   
12   114101516  20140528T000000   310000.0         3       1.00         1430   
13  6054650070  20141007T000000   400000.0         3       1.75         1370   
14  1175000570  20150312T000000   530000.0         5       2.00         1810   
15  9297300055  20150124T000000   650000.0         4       3.00         2950   
16  1875500060  20140731T000000   395000.0         3       2.00         1890   
17  6865200140  20140529T000000   485000.0         4       1.00         1600   
18    16000397  20141205T000000   189000.0         2       1.00         1200   
19  7983200060  20150424T000000   230000.0         3       1.00         1250   

    sqft_lot  floors  waterfront  view  ...  yr_built  yr_renovated  zipcode  \
0       5650     1.0           0     0  ...      1955             0    98178   
1       7242     2.0           0     0  ...      1951          1991    98125   
2      10000     1.0           0     0  ...      1933             0    98028   
3       5000     1.0           0     0  ...      1965             0    98136   
4       8080     1.0           0     0  ...      1987             0    98074   
5     101930     1.0           0     0  ...      2001             0    98053   
6       6819     2.0           0     0  ...      1995             0    98003   
7       9711     1.0           0     0  ...      1963             0    98198   
8       7470     1.0           0     0  ...      1960             0    98146   
9       6560     2.0           0     0  ...      2003             0    98038   
10      9796     1.0           0     0  ...      1965             0    98007   
11      6000     1.0           0     0  ...      1942             0    98115   
12     19901     1.5           0     0  ...      1927             0    98028   
13      9680     1.0           0     0  ...      1977             0    98074   
14      4850     1.5           0     0  ...      1900             0    98107   
15      5000     2.0           0     3  ...      1979             0    98126   
16     14040     2.0           0     0  ...      1994             0    98019   
17      4300     1.5           0     0  ...      1916             0    98103   
18      9850     1.0           0     0  ...      1921             0    98002   
19      9774     1.0           0     0  ...      1969             0    98003   

        lat     long  sqft_living15  sqft_lot15  delta_renovated  yr_sale  \
0   47.5112 -122.257           1340        5650                0     2014   
1   47.7210 -122.319           1690        7639               40     2014   
2   47.7379 -122.233           2720        8062                0     2015   
3   47.5208 -122.393           1360        5000                0     2014   
4   47.6168 -122.045           1800        7503                0     2015   
5   47.6561 -122.005           4760      101930                0     2014   
6   47.3097 -122.327           2238        6819                0     2014   
7   47.4095 -122.315           165

**1.4** Удалите признаки date, zipcode, lat, long

In [5]:
df.drop(columns=['date','zipcode','long'],inplace=True)
df.head(10)

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0  7129300520   221900.0         3       1.00         1180      5650     1.0   
1  6414100192   538000.0         3       2.25         2570      7242     2.0   
2  5631500400   180000.0         2       1.00          770     10000     1.0   
3  2487200875   604000.0         4       3.00         1960      5000     1.0   
4  1954400510   510000.0         3       2.00         1680      8080     1.0   
5  7237550310  1225000.0         4       4.50         5420    101930     1.0   
6  1321400060   257500.0         3       2.25         1715      6819     2.0   
7  2008000270   291850.0         3       1.50         1060      9711     1.0   
8  2414600126   229500.0         3       1.00         1780      7470     1.0   
9  3793500160   323000.0         3       2.50         1890      6560     2.0   

   waterfront  view  condition  ...  sqft_above  sqft_basement  yr_built  \
0           0     0          3  ...        1180              0      1955   
1           0     0          3  ...        2170            400      1951   
2           0     0          3  ...         770              0      1933   
3           0     0          5  ...        1050            910      1965   
4           0     0          3  ...        1680              0      1987   
5           0     0          3  ...        3890           1530      2001   
6           0     0          3  ...        1715              0      1995   
7           0     0          3  ...        1060              0      1963   
8           0     0          3  ...        1050            730      1960   
9           0     0          3  ...        1890              0      2003   

   yr_renovated      lat  sqft_living15  sqft_lot15  delta_renovated  yr_sale  \
0             0  47.5112           1340        5650                0     2014   
1          1991  47.7210           1690        7639               40     2014   
2             0  47.7379           2720        8062                0     2015   
3             0  47.5208           1360        5000                0     2014   
4             0  47.6168           1800        7503                0     2015   
5             0  47.6561           4760      101930                0     2014   
6             0  47.3097           2238        6819                0     2014   
7             0  47.4095           1650        9711                0     2015   
8             0  47.5123           1780        8113                0     2015   
9             0  47.3684           2390        7570                0     2015   

  month_sale  
0        Okt  
1        Dec  
2        Feb  
3        Dec  
4        Feb  
5        May  
6        Jun  
7        Jan  
8        Apr  
9        Mar  

[10 rows x 21 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   id               21613 non-null  int64  
 1   price            21613 non-null  float64
 2   bedrooms         21613 non-null  int64  
 3   bathrooms        21613 non-null  float64
 4   sqft_living      21613 non-null  int64  
 5   sqft_lot         21613 non-null  int64  
 6   floors           21613 non-null  float64
 7   waterfront       21613 non-null  int64  
 8   view             21613 non-null  int64  
 9   condition        21613 non-null  int64  
 10  grade            21613 non-null  int64  
 11  sqft_above       21613 non-null  int64  
 12  sqft_basement    21613 non-null  int64  
 13  yr_built         21613 non-null  int64  
 14  yr_renovated     21613 non-null  int64  
 15  lat              21613 non-null  float64
 16  sqft_living15    21613 non-null  int64  
 17  sqft_lot15  

**Условие 2: 2 задача**
Создайте датафрейм с клиентами:

clients = pd.DataFrame({
'client_id': [1459, 4684, 3498, 3942, 4535, 2308, 2866, 2765, 1472, 4236, 2295,
939, 3840, 280, 20, 4332, 3475, 4213, 3113, 4809, 2134, 2242,
2068, 4929, 1384, 1589, 3317, 2260, 1727, 1764, 1611, 1474],
'house_id': [8965450190, 6823100225, 5104540330, 2131701075, 1522700060,
1189000207, 6821600300, 7137950720, 9510920050, 6131600255,
5428000070, 1788800910, 8100400160, 3123049142, 6306800010,
5083000375, 7920100025, 1951600150, 809001400, 339600110,
1622049154, 1099600250, 8563000110, 2768100205, 3995700435,
8861700030, 3303980210, 7731100066, 8146100580, 825069097,
3889100029, 9524100196]
})

In [7]:
clients = pd.DataFrame({ 'client_id': [1459, 4684, 3498, 3942, 4535, 2308, 2866, 2765, 1472, 4236, 2295, 939, 3840, 280, 20, 4332, 3475, 4213, 3113, 4809, 2134, 2242, 2068, 4929, 1384, 1589, 3317, 2260, 1727, 1764, 1611, 1474], 'house_id': [8965450190, 6823100225, 5104540330, 2131701075, 1522700060, 1189000207, 6821600300, 7137950720, 9510920050, 6131600255, 5428000070, 1788800910, 8100400160, 3123049142, 6306800010, 5083000375, 7920100025, 1951600150, 809001400, 339600110, 1622049154, 1099600250, 8563000110, 2768100205, 3995700435, 8861700030, 3303980210, 7731100066, 8146100580, 825069097, 3889100029, 9524100196] })
clients

client_id    house_id
0        1459  8965450190
1        4684  6823100225
2        3498  5104540330
3        3942  2131701075
4        4535  1522700060
5        2308  1189000207
6        2866  6821600300
7        2765  7137950720
8        1472  9510920050
9        4236  6131600255
10       2295  5428000070
11        939  1788800910
12       3840  8100400160
13        280  3123049142
14         20  6306800010
15       4332  5083000375
16       3475  7920100025
17       4213  1951600150
18       3113   809001400
19       4809   339600110
20       2134  1622049154
21       2242  1099600250
22       2068  8563000110
23       4929  2768100205
24       1384  3995700435
25       1589  8861700030
26       3317  3303980210
27       2260  7731100066
28       1727  8146100580
29       1764   825069097
30       1611  3889100029
31       1474  9524100196

**2.1** Присоедините к таблице clients данные по домам через метод join

In [8]:
df.shape

(21613, 21)

In [9]:
clients.shape

(32, 2)

In [10]:
clients_id = clients.set_index('house_id')
clients_id.head()

client_id
house_id             
8965450190       1459
6823100225       4684
5104540330       3498
2131701075       3942
1522700060       4535

In [11]:
df_id=df.set_index('id')
df_id

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
id                                                                         
7129300520  221900.0         3       1.00         1180      5650     1.0   
6414100192  538000.0         3       2.25         2570      7242     2.0   
5631500400  180000.0         2       1.00          770     10000     1.0   
2487200875  604000.0         4       3.00         1960      5000     1.0   
1954400510  510000.0         3       2.00         1680      8080     1.0   
...              ...       ...        ...          ...       ...     ...   
263000018   360000.0         3       2.50         1530      1131     3.0   
6600060120  400000.0         4       2.50         2310      5813     2.0   
1523300141  402101.0         2       0.75         1020      1350     2.0   
291310100   400000.0         3       2.50         1600      2388     2.0   
1523300157  325000.0         2       0.75         1020      1076     2.0   

            waterfront  view  condition  grade  sqft_above  sqft_basement  \
id                                                                          
7129300520           0     0          3      7        1180              0   
6414100192           0     0          3      7        2170            400   
5631500400           0     0          3      6         770              0   
2487200875           0     0          5      7        1050            910   
1954400510           0     0          3      8        1680              0   
...                ...   ...        ...    ...         ...            ...   
263000018            0     0          3      8        1530              0   
6600060120           0     0          3      8        2310              0   
1523300141           0     0          3      7        1020              0   
291310100            0     0          3      8        1600              0   
1523300157           0     0          3      7        1020              0   

            yr_built  yr_renovated      lat  sqft_living15  sqft_lot15  \
id                                                                       
7129300520      1955             0  47.5112           1340        5650   
6414100192      1951          1991  47.7210           1690        7639   
5631500400      1933             0  47.7379           2720        8062   
2487200875      1965             0  47.5208           1360        5000   
1954400510      1987             0  47.6168           1800        7503   
...              ...           ...      ...            ...         ...   
263000018       2009             0  47.6993           1530        1509   
6600060120      2014             0  47.5107           1830        7200   
1523300141      2009             0  47.5944           1020        2007   
291310100       2004             0  47.5345           1410        1287   
1523300157      2008             0  47.5941           1020        1357   

            delta_renovated yr_sale month_sale  
id                                              
7129300520                0    2014        Okt  
6414100192               40    2014        Dec  
5631500400                0    2015        Feb  
2487200875                0    2014        Dec  
1954400510                0    2015        Feb  
...                     ...     ...        ...  
263000018                 0    2014        May  
6600060120                0    2015        Feb  
1523300141                0    2014        Jun  
291310100                 0    2015        Jan  
1523300157                0    2014        Okt  

[21613 rows x 20 columns]

In [12]:
df_join = clients_id.join(df_id)
df_join

client_id      price  bedrooms  bathrooms  sqft_living  sqft_lot  \
339600110        4809   395000.0         3       2.50         1610      3755   
809001400        3113   925000.0         3       1.00         1630      3200   
825069097        1764   770000.0         3       2.50         2650     40705   
1099600250       2242   260000.0         3       1.75         1710      6400   
1189000207       2308   387000.0         2       2.50         1170      1394   
1522700060       4535   518000.0         4       2.75         2520     14021   
1622049154       2134   289900.0         3       1.75         1899     11325   
1788800910        939   190000.0         3       1.00         1200     10458   
1951600150       4213   180000.0         3       1.00         1610      8500   
2131701075       3942   420000.0         3       1.75         1720      5000   
2768100205       4929   519000.0         4       2.50         1950      2617   
3123049142        280   452000.0         3       2.25         2600     14810   
3303980210       3317  1115000.0         4       3.75         4040     14212   
3889100029       1611   810000.0         3       2.50         2670     10481   
3995700435       1384   265000.0         4       3.00         1940      8170   
5083000375       4332   170000.0         3       1.00         1310      9529   
5083000375       4332   235000.0         3       1.00         1310      9529   
5104540330       3498   679000.0         4       2.50         3680      7236   
5428000070       2295   770000.0         5       3.50         4750      8234   
6131600255       4236   202500.0         3       2.00         1540      8316   
6306800010         20   436472.0         4       2.50         2692      8392   
6821600300       2866   886000.0         3       2.25         2380      6000   
6823100225       4684   700000.0         4       1.75         1870      6000   
7137950720       2765   339100.0         4       2.50         2350     10655   
7731100066       2260   545000.0         3       1.00         1510      5000   
7920100025       3475   450000.0         2       1.00          740      5100   
8100400160       3840   700000.0         3       2.25         2330     11424   
8146100580       1727   765000.0         3       1.00         1220      7585   
8563000110       2068   427000.0         4       1.75         1460      9750   
8861700030       1589   510000.0         3       1.50         2400     10275   
8965450190       1459   295000.0         3       2.50         1500      3060   
9510920050       1472   725000.0         3       2.50         2980     16996   
9524100196       1474   239000.0         2       1.50          680       772   

            floors  waterfront  view  condition  ...  sqft_above  \
339600110      2.0           0     0          3  ...        1610   
809001400      1.5           0     0          4  ...        1630   
825069097      2.0           0     0          3  ...        2650   
1099600250     1.0           0     0          4  ...        1240   
1189000207     2.0           0     0          3  ...        1170   
1522700060     2.0           0     0          3  ...        2520   
1622049154     2.0           0     0          3  ...        1899   
1788800910     1.0           0     0          4  ...        1200   
1951600150     1.5           0     0          3  ...        1610   
2131701075     1.5           0     0          3  ...        1720   
2768100205     1.5           0     0          4  ...        1250   
3123049142     1.0           0     2          4  ...        1490   
3303980210     2.0           0     0          3  ...        4040   
3889100029     2.0           0     0          3  ...        2670   
3995700435     1.0           0     0          4  ...        1940   
5083000375     1.0           0     0          3  ...        1310   
5083000375     1.0           0     0          3  ...        1310   
5104540330     2.0           0     0          3  ...        3680   

In [13]:
df_join.shape

(33, 21)

**2.2** Присоедините к таблице clients данные по домам через метод merge

Это нужно, чтобы понимать, какие дома покупались клиентами
house_id - это индексы датафрейма с домами

In [14]:
df_merge=clients.merge(df,left_on='house_id', right_on='id')
df_merge.head()

client_id    house_id          id     price  bedrooms  bathrooms  \
0       1459  8965450190  8965450190  295000.0         3       2.50   
1       4684  6823100225  6823100225  700000.0         4       1.75   
2       3498  5104540330  5104540330  679000.0         4       2.50   
3       3942  2131701075  2131701075  420000.0         3       1.75   
4       4535  1522700060  1522700060  518000.0         4       2.75   

   sqft_living  sqft_lot  floors  waterfront  ...  sqft_above  sqft_basement  \
0         1500      3060     2.0           0  ...        1500              0   
1         1870      6000     1.0           0  ...        1670            200   
2         3680      7236     2.0           0  ...        3680              0   
3         1720      5000     1.5           0  ...        1720              0   
4         2520     14021     2.0           0  ...        2520              0   

   yr_built  yr_renovated      lat  sqft_living15  sqft_lot15  \
0      1994             0  47.5605           2700        7734   
1      1949             0  47.6435           1710        6000   
2      2006             0  47.3543           3310        7180   
3      1932          2009  47.7380           1410        8300   
4      1999             0  47.7344           2330       14007   

   delta_renovated  yr_sale  month_sale  
0                0     2015         Feb  
1                0     2015         Apr  
2                0     2015         May  
3               77     2014         Dec  
4                0     2014         Jun  

[5 rows x 23 columns]

In [15]:
df_merge.shape

(33, 23)

**Условие 3: 3 задача**
Составьте несколько сводных таблиц

**3.1** Найдите среднюю стоимость домов в зависимости от количества спален
Отсортируйте от меньшей стоимости к большей

In [16]:
df.groupby('bedrooms').agg({'price':'mean'}).sort_values('price')


price
bedrooms              
1         3.176429e+05
2         4.013727e+05
0         4.095038e+05
3         4.662321e+05
11        5.200000e+05
4         6.354195e+05
33        6.400000e+05
5         7.865998e+05
10        8.193333e+05
6         8.255206e+05
9         8.939998e+05
7         9.511847e+05
8         1.105077e+06

**3.2** Найдите минимальную, среднюю и максимальную стоимости домов в зависимости от состояния дома

In [17]:
round(df.groupby('condition').agg({'price':['min','max','mean']}),2)

price                      
                min        max       mean
condition                                
1           78000.0  1500000.0  334431.67
2           80000.0  2555000.0  327287.15
3           75000.0  7062500.0  542012.58
4           89000.0  7700000.0  521200.39
5          110000.0  3650000.0  612418.09

**3.3** Постройте таблицу с подсчетом количества домов в данных в зависимости от вида на набережную и оценкой вида

In [18]:
df.pivot_table(index='waterfront',columns='view',
              values='id', aggfunc='count',
              fill_value=0)

view            0    1    2    3    4
waterfront                           
0           19489  331  955  491  184
1               0    1    8   19  135

**3.4** Каких домов в зависимости от этажности и количества спален больше?

In [19]:
pd.crosstab(index=df['floors'],columns=df['bedrooms'])

bedrooms  0    1     2     3     4    5    6   7   8   9   10  11  33
floors                                                               
1.0        4  162  1951  5455  2383  605  104   9   5   0   1   0   1
1.5        0   21   182   786   698  185   30   7   1   0   0   0   0
2.0        6   12   497  3118  3682  775  119  19   6   4   2   1   0
2.5        0    1     5    56    58   23   14   2   0   2   0   0   0
3.0        2    3   123   405    61   13    5   1   0   0   0   0   0
3.5        1    0     2     4     0    0    0   0   1   0   0   0   0

In [20]:
pd.crosstab(index=df['floors'],columns=df['bedrooms']).max()

bedrooms
0        6
1      162
2     1951
3     5455
4     3682
5      775
6      119
7       19
8        6
9        4
10       2
11       1
33       1
dtype: int64

**Вывод:** Больше всего одноэтажных домов с тремя спальнями 5455

**3.5** Постройте таблицу с подсчетом медианной стоимости домов в данных в зависимости от состояния дома и оценки дома

In [21]:
round(df.pivot_table(index='condition',columns='grade',
              values='price', aggfunc='mean',
              fill_value=0
              ),2)

grade          1       3       4          5          6          7          8   \
condition                                                                       
1          142000       0  150000  280833.33  303813.64  334416.67  932500.00   
2               0  280000  179600  206166.67  269483.90  342698.52  421996.15   
3               0   75000  210000  246993.46  290651.85  385666.79  510102.77   
4               0       0  243905  245299.01  318731.57  412923.07  589527.30   
5               0  262000       0  271127.59  310140.86  479667.75  736538.05   

grade              9           10          11          12          13  
condition                                                              
1                0.00        0.00        0.00        0.00        0.00  
2           715000.00  1752500.00        0.00        0.00        0.00  
3           726587.80  1012934.13  1414401.44  2138601.08  3338636.36  
4           889664.52  1176284.59  1839916.43  2467961.54  5750000.00  
5          1123419.90  1735830.36  2238477.27  2290000.00        0.00